src/plugins/data/public/search/session/session_service.ts

```diff
@@ -263,7 +289,32 @@ export class SessionService {
 
     // if we are still interested in this result
     if (this.getSessionId() === sessionId) {
-      this.state.transitions.store();
+      this.state.transitions.store(searchSessionSavedObject);
+    }
+  }
+
+  /**
+   * Change user-facing name of a current session
+   * @param newName - new session name
+   */
+  public async renameCurrentSession(newName: string) {
+    const sessionId = this.getSessionId();
+    if (sessionId && this.state.get().isStored) {
+      let renamed = false;
+      try {
+        await this.sessionsClient.rename(sessionId, newName);
+        renamed = true;
+      } catch (e) {
+        this.toastService?.addError(e, {
+          title: i18n.translate('data.searchSessions.sessionService.sessionEditNameError', {
+            defaultMessage: 'Failed to edit name of the search session',
+          }),
+        });
+      }
+
+      if (renamed && sessionId === this.getSessionId()) {
+        await this.refreshSearchSessionSavedObject();
```

Comment:
`this.sessionsClient.rename(sessionId, newName)` already gets the updated saved object, in response to the update, I think. But it doesn't return it.
So you don't have to refetch it with an extra network call IMO.

Reply:
![image](https://user-images.githubusercontent.com/3016806/111504884-bb091480-8750-11eb-8291-dbb8ee015237.png)
Both requests have the same response format it seems

Reply:
Isn't the `update` request returns only the updated fields? 
I didn't really want to _merge_ partial SO on a client to reduce possible bug surface. 

I also plan to use `this.refreshSearchSessionSavedObject` for periodic fetching of SO in https://github.com/elastic/kibana/issues/92785

Reply:
Gotcha.

However why would you need to refetch the SO for https://github.com/elastic/kibana/issues/92785 ?
It doesn't sound like we need to be polling more things :)

Reply:
In design there, we also show "completed" time on a completed state. [figma](https://www.figma.com/file/FD2M7MUpLScJKOyYjfbmev/Search-Sessions?node-id=1549%3A0)

But I think would be less confusing if we'd eventually show REAL server-side completed time and not a client-side one: https://www.figma.com/file/FD2M7MUpLScJKOyYjfbmev/Search-Sessions?node-id=1549%3A0. Not 100% sure just yet if it worth it.

Reply:
I don't think it's worth it.
While the session is running, I would either show the time where the client sides decides a session is done or not at all.

Reply:
Great! thanks for the input. I'd start with pure client-side in that pr and then we will see if something else is needed 👍